server/src/data/getApps.js

```diff
@@ -5,25 +5,51 @@ const debug = require('debug')('apphub:server:data:getApps')
  * @param {object} params
  * @param {string} params.status Which status to get the apps for, example APPROVED AppTypes in src/enum
  * @param {string} params.languageCode The language code for which language to use when fetching translations
- * @param {string} params.channel Name of the channel to filter apps by, for example Stable, Development, Canary is commonly used
+ * @param {array} params.channels Channels to filter apps by, for example Stable, Development, Canary is commonly used
+ * @param {array} params.types Types to filter apps by
+ * @param {string} params.query Search query for app names
  * @returns {Promise<Array>}
- *
  */
-const getApps = ({ status, languageCode, channel }, knex) => {
+const getApps = (
+    { status, languageCode, channels = [], types = [], query },
+    knex
+) => {
     debug('status:', status)
     debug('languageCode:', languageCode)
-    debug('channel:', channel)
+    debug('channels:', channels)
 
     return knex('apps_view')
         .select()
         .where(builder => {
             builder.where('status', status)
             builder.where('language_code', languageCode)
 
-            if (channel) {
-                builder.where('channel_name', channel)
+            if (channels.length > 0) {
+                builder.where(builder => {
+                    builder.where('channel_name', channels[0])
+                    channels.slice(1).forEach(channel => {
+                        builder.orWhere('channel_name', channel)
+                    })
+                })
+            }
+
+            if (types.length > 0) {
+                builder.where(builder => {
+                    builder.where('type', types[0])
+                    types.slice(1).forEach(type => {
+                        builder.orWhere('type', type)
+                    })
+                })
+            }
```

Comment:
These should be case-insensitive, are they? - I haven't tested, but it appears that `?channels=stable` won't work (because it expects `Stable`)

Reply:
`Stable` is how the channel name is stored in the DB currently. I'll make a migration to rename it to `stable` :+1:

Reply:
OK 👍 - if you want to do that in a separate change that could be OK too

Reply:
[This commit](https://github.com/dhis2/app-hub/pull/431/commits/f9b90f543276d27dcbe8c14bbef44f706c98f386) has the changes for renaming channels.

Should I also guarantee case-insensitivity in the API by doing something like this?
```js
builder.where(builder => {
    builder.where('channel_name', channels[0].toLowerCase())
    channels.slice(1).forEach(channel => {
        builder.orWhere('channel_name', channel.toLowerCase())
    })
})
```

Reply:
> Should I also guarantee case-insensitivity in the API by doing something like this? 

I vote yes!

Reply:
Done :heavy_check_mark: